### PR TITLE
[HDRP] Add custom post process error in the UI when the component is not added in the global settings

### PR DESCRIPTION
**Pull request by alelievr**
### Purpose of this PR
Add custom post-process error in the UI when the component is not added in the global settings:

image

---
### Testing status
Tested that multiple levels of inheritance are correctly supported in volumes. Checked that other volumes are not impacted by this change.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/PostProcessing/CustomPostProcessVolumeComponentEditor.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+using UnityEngine.Rendering.HighDefinition;
+using UnityEngine.Rendering;
+
+namespace UnityEditor.Rendering.HighDefinition
+{
+    /// <summary>
+    /// Base class to inherit to create custom post process volume editors.
+    /// </summary>
+    [CanEditMultipleObjects]
+    [VolumeComponentEditor(typeof(CustomPostProcessVolumeComponent))]
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.high-definition/Editor/PostProcessing/CustomPostProcessVolumeComponentEditor.cs:11]:
```suggestion
    [CustomEditor(typeof(CustomPostProcessVolumeComponent))]
```

VolumeComponent Editor is deprecated, see `com.unity.render-pipelines.core\Editor\Deprecated.cs`

File: com.unity.render-pipelines.high-definition/Editor/PostProcessing/CustomPostProcessVolumeComponentEditor.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        /// <remarks>
+        /// You can safely override this method and not call <c>base.OnInspectorGUI()</c> unless you
+        /// want Unity to display all the properties from the <see cref="VolumeComponent"/>
+        /// automatically.
+        /// </remarks>
+        public override void OnInspectorGUI()
+        {
+            if (!HDRenderPipelineGlobalSettings.instance?.IsCustomPostProcessRegistered(target.GetType()) ?? false)
+            {
+                HDEditorUtils.GlobalSettingsHelpBox("The custom post process is not registered in the Global Settings.",
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.high-definition/Editor/PostProcessing/CustomPostProcessVolumeComponentEditor.cs:26]: move the string to a static const string into an inner class Styles

**Review by alex-vazquez — CHANGES_REQUESTED** [review]

File: com.unity.render-pipelines.high-definition/Editor/PostProcessing/CustomPostProcessVolumeComponentEditor.cs (hunk 1 above)
Comment (alelievr) [reply com.unity.render-pipelines.high-definition/Editor/PostProcessing/CustomPostProcessVolumeComponentEditor.cs:11]: Yes, I did the changes with a completely outdated version of the repo.
Thanks for the suggestion block!

Comment (alelievr) [reply com.unity.render-pipelines.high-definition/Editor/PostProcessing/CustomPostProcessVolumeComponentEditor.cs]: Hum, this is breaking my fix, I need to investigate why. Maybe inheritance isn't supported by default with the CustomEditor attribute :/

Comment (alex-vazquez) [reply com.unity.render-pipelines.high-definition/Editor/PostProcessing/CustomPostProcessVolumeComponentEditor.cs]: Maybe, you are missing the parameter after the  type, to be true, to be used by the children?

Comment (alelievr) [reply com.unity.render-pipelines.high-definition/Editor/PostProcessing/CustomPostProcessVolumeComponentEditor.cs]: Ah yes, forgot about that. It works now, thanks!

File: com.unity.render-pipelines.high-definition/Editor/PostProcessing/CustomPostProcessVolumeComponentEditor.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    /// <summary>
+    /// Base class to inherit to create custom post process volume editors.
+    /// </summary>
+    [CanEditMultipleObjects]
+    [CustomEditor(typeof(CustomPostProcessVolumeComponent), true)]
+    public class CustomPostProcessVolumeComponentEditor : VolumeComponentEditor
+    {
+        internal static class Styles
+        {
+            public static readonly string customPostProcessNotInGlobalSettingsText = "The custom post process is not registered in the Global Settings.";
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.high-definition/Editor/PostProcessing/CustomPostProcessVolumeComponentEditor.cs:16]: Nice, thank you :)

**Review by alex-vazquez — APPROVED** [review]
Looks good! Thank you for your code changes.

**Review by AlixMi — APPROVED** [review]
Ok that's good and will help a lot for custom post process creation. 
Maybe I would say "This Custom Postprocess is not registered [...]" instead of "The custom postprocess[...]" we tend to have capital letter around important work I think 
Thanks!

**Comment by alelievr** [comment]
@AlixMi Yeah, I updated the message.

**Review by sebastienlagarde — APPROVED** [review]

**Review by TomasKiniulis — APPROVED** [review]
Looks good. Tested with multiple Custom Post Process assigned/unassigned to Global Settings with a multiple locked Inspectors and separate volumes. Checked switch to builtin and back to HDRP - error message persists or is not shown when assigned.

**Pull request merged**